From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <Blake.Galbreath@GMail.com>
Date: Sat, 15 Aug 2020 09:32:00 -0500
Subject: [PATCH] Fix incorrect return for WorldServer#addAllEntitiesSafely


diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 79c3cbc646b232057dd33418b56709a594974e41..a296401df96c7400e4ad6d3e91223b0e5cdfce5b 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -1262,7 +1262,11 @@ public class WorldServer extends World implements GeneratorAccessSeed {
         if (entity.cp().anyMatch(this::isUUIDTaken)) {
             return false;
         } else {
-            return this.addAllEntities(entity, reason); // CraftBukkit
+            // Paper start - this method is _only_ designed to return false on duplicate uuids
+            // fixes issues with things such as a plugin cancelling spawn of a /summon
+            this.addAllEntities(entity, reason); // CraftBukkit
+            return true;
+            // Paper end
         }
     }
 
